Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -871,7 +990,7 @@ public boolean onTouch(View v, MotionEvent ev) {
 	 */
 	public void closeSession() {
 		if (prefHelper_.getSmartSession()) {
-			if (keepAlive_) {
+			if (keepAlive_ && !isBranchAppInstantiated) { //No need to check for keepAlive_ for auto session management
```

Comment:
If the new session management is in use, closeSession should have no effect, right? So we should return here immediately, otherwise, it will go on and call `executeClose()` down there, no?